test/fs/test_fs_eisdir.c

```diff
@@ -0,0 +1,36 @@
+#include <unistd.h>
+#include <sys/stat.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <string.h>
+#include <assert.h>
+
+#if defined(__EMSCRIPTEN__)
+#include <emscripten.h>
+#endif
+
+void makedir(const char *dir) {
+  int rtn = mkdir(dir, 0777);
+  assert(rtn == 0);
+}
+
+void changedir(const char *dir) {
+  int rtn = chdir(dir);
+  assert(rtn == 0);
+}
+
+void setup() {
+#if defined(__EMSCRIPTEN__) && defined(NODEFS)
+  makedir("working");
+  EM_ASM(FS.mount(NODEFS, { root: '.' }, 'working'));
+  changedir("working");
+#endif
+}
```

Comment:
This setup code should no longer be needed now that we have `setup_nodefs_test()` (#23167)

Reply:
I was waiting for https://github.com/emscripten-core/emscripten/pull/23177 too.

Reply:
OK, thats landed now.